firstvoices/backend/management/commands/copy_site.py

```diff
@@ -0,0 +1,581 @@
+        variant.save()
+
+    ignored_characters = list(IgnoredCharacter.objects.filter(site=source_site))
+    for ignored_character in ignored_characters:
+        ignored_character.id = uuid.uuid4()
+        ignored_character.site = target_site
+        ignored_character.save()
+
+    return character_map
+
+
+def copy_alphabet(source_site, target_site):
+    alphabets = list(Alphabet.objects.filter(site=source_site))
+    for alphabet in alphabets:
+        alphabet.id = uuid.uuid4()
+        alphabet.site = target_site
+        alphabet.save()
+
+
+def copy_categories_and_return_map(source_site, target_site, user):
+    category_map = {}
+
+    # Removing auto-generated categories
+    Category.objects.filter(site=target_site).delete()
+    # Copy over all the parent categories
+    categories = list(
+        Category.objects.filter(site=source_site, children__isnull=False).distinct()
+    )
+    for category in categories:
+        child_categories = list(category.children.all())
+        for child_category in child_categories:
+            source_category_id = child_category.id
+            target_category_id = uuid.uuid4()
+            category_map[source_category_id] = target_category_id
+
+            child_category.id = target_category_id
+            child_category.site = target_site
+            child_category.created_by = user
+            child_category.last_modified_by = user
+            child_category.save()
+
+        source_category_id = category.id
+        target_category_id = uuid.uuid4()
+        category_map[source_category_id] = target_category_id
+
+        category.id = target_category_id
+        category.site = target_site
+        category.created_by = user
+        category.last_modified_by = user
+        category.save()
+
+        for child_category in child_categories:
+            child_category.parent = category
+            child_category.created_by = user
+            child_category.last_modified_by = user
+            child_category.save()
+
+    # Updated category list, copying over rest of categories
+    target_site_copied_over_categories = list(
+        Category.objects.filter(site=target_site).values_list("title", flat=True)
+    )
+    categories = list(
+        Category.objects.filter(site=source_site)
+        .exclude(title__in=target_site_copied_over_categories)
+        .distinct()
+    )
+    for category in categories:
+        source_category_id = category.id
+        target_category_id = uuid.uuid4()
+        category_map[source_category_id] = target_category_id
+
+        category.id = target_category_id
+        category.site = target_site
+        category.created_by = user
+        category.last_modified_by = user
+        category.save()
+
+    return category_map
+
+
+def copy_audio_and_speakers_and_return_map(source_site, target_site, logger):
+    audio_map = {}
+
+    audio_files = list(Audio.objects.filter(site=source_site))
+    for audio_file in audio_files:
```

Comment:
[minor] for clarity I think "audio_file" should be "audio_model" so they don't get confused with the content of an audio model